### PR TITLE
change status_format to return an existing string as-is

### PR DESCRIPTION
**Pull request by nweisenfeld**
# Description

Right now, if status_format is called with a value that's a str, then it will recurse forever.

The initial isinstance(value,Iterable) is true for the iterable string, and it calls itself recursively for each character, which is also an iterable string, which calls status_format ad infinitum.

status_format() takes value and returns a friendly formatted string, so the change here simply returns an incoming string as-is.

My personal rationale here is so that I can override get_value() in a layer and co-opt the status bar to display, for instance, detailed information about a point when the mouse hovers over it, rather than simply displaying a coordinate and an index of the point.  

While my status line usage can certainly be a point of discussion, the change here boils down to fixing a bug where this function will never return.

## Type of change

- [X] Bug-fix (non-breaking change which fixes an issue)

# References

# How has this been tested?

- [X] passes all tests on my system except cli, which even the release 0.4.4 fails

## Final checklist:
- [X] I have added tests that prove my fix is effective or that my feature works - added test_status.py to _tests

### CODE REVIEW

File: napari/_tests/test_status.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import numpy as np
```
Comment (sofroniewn) [inline napari/_tests/test_status.py]: Can you move this test to `napari/utils/_tests/test_status.py`?

**Review by sofroniewn — APPROVED** [review]
I've requested the test location to move, but otherwise it looks good to me and i'm happy to merge.

> My personal rationale here is so that I can override get_value() in a layer and co-opt the status bar to display, for instance, detailed information about a point when the mouse hovers over it, rather than simply displaying a coordinate and an index of the point.

Hmm, if you wanted to open a dedicate issue I'd be happy to discuss this more to make it easier for you and other users. I know @VolkerH is very interested too.

File: napari/_tests/test_status.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+from napari.utils.status_messages import status_format
+
+
+def test_status_format(make_napari_viewer):
```
Comment (sofroniewn) [inline napari/_tests/test_status.py]:
```suggestion
def test_status_format():
```
This test doesn't need the napari viewer

File: napari/_tests/test_status.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    """ test various formatting cases embodied in utils.status_messages.status_format """
+
+    values = np.array(
```
Comment (sofroniewn) [inline napari/_tests/test_status.py]: Can you also make this test look more like https://github.com/napari/napari/blob/133ab7f70803b7db26f772335b30fb96e4a44b02/napari/utils/_tests/test_misc.py#L21-L35 where we have `input, expected` and then the test just becomes

```python
assert status_format(input) == expected
```

**Comment by sofroniewn** [comment]
Left a couple extra comments on the test, thanks again for the bug fix!!

**Review by jni — APPROVED** [review]
Thanks @nweisenfeld!

I agree with @sofroniewn's minor update requests and otherwise this is ready to merge for me. 👍

File: napari/_tests/test_status.py (hunk 1 above)
Comment (nweisenfeld) [reply napari/_tests/test_status.py]: ack, yes - I should have looked more closely at the organization.

File: napari/_tests/test_status.py (hunk 3 above)
Comment (nweisenfeld) [reply napari/_tests/test_status.py]: thanks, changes made!

**Comment by nweisenfeld** [comment]
@jni @sofroniewn this is my first PR to Napari, so let me know the workflow.  This is good to go as far as I'm concerned - I've addressed the comments.

**Comment by sofroniewn** [comment]
Tests look great now!! Thanks so much for updating them I'll merge this now :-)

**Pull request merged**